File: src/main/java/com/learnershi/rclasssocket/repository/RevealRepository.java
package com.learnershi.rclasssocket.repository;

import com.learnershi.rclasssocket.entity.Reveal;
import org.springframework.data.mongodb.repository.ReactiveMongoRepository;
import reactor.core.publisher.Mono;

public interface RevealRepository extends ReactiveMongoRepository<Reveal, String> {
    Mono<Reveal> findByClassRoomIdAndTabIndexAndPageIndex(String classRoomId, Integer tabIndex, Integer pageIndex);
}

File: src/main/java/com/learnershi/rclasssocket/repository/RevealRepository.kt
package com.learnershi.rclasssocket.repository

import com.learnershi.rclasssocket.entity.Reveal
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import reactor.core.publisher.Mono

interface RevealRepository : ReactiveMongoRepository<Reveal?, String?> {
    fun findByClassRoomIdAndTabIndexAndPageIndex(classRoomId: String?, tabIndex: Int?, pageIndex: Int?): Mono<Reveal?>
}